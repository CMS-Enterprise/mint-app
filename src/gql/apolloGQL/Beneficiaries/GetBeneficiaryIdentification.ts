import { gql } from '@apollo/client';

export default gql(/* GraphQL */ `
  query GetBeneficiaryIdentification($id: UUID!) {
    modelPlan(id: $id) {
      id
      modelName
      beneficiaries {
        id
        beneficiaries
        diseaseSpecificGroup
        beneficiariesOther
        beneficiariesNote
        treatDualElligibleDifferent
        treatDualElligibleDifferentHow
        treatDualElligibleDifferentNote
        excludeCertainCharacteristics
        excludeCertainCharacteristicsCriteria
        excludeCertainCharacteristicsNote
      }
    }
  }
`);
