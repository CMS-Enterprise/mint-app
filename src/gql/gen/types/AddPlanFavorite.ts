/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
