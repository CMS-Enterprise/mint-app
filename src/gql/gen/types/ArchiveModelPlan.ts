/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: ArchiveModelPlan
// ====================================================

export interface ArchiveModelPlan_updateModelPlan {
  __typename: "ModelPlan";
  id: UUID;
}

export interface ArchiveModelPlan {
  updateModelPlan: ArchiveModelPlan_updateModelPlan;
}

export interface ArchiveModelPlanVariables {
  id: UUID;
  archived: boolean;
}
