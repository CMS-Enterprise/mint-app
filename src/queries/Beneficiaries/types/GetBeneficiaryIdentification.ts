/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { BeneficiariesType, TriStateAnswer } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: GetBeneficiaryIdentification
// ====================================================

export interface GetBeneficiaryIdentification_modelPlan_beneficiaries {
  __typename: "PlanBeneficiaries";
  id: UUID;
  beneficiaries: BeneficiariesType[];
  beneficiariesOther: string | null;
  beneficiariesNote: string | null;
  treatDualElligibleDifferent: TriStateAnswer | null;
  treatDualElligibleDifferentHow: string | null;
  treatDualElligibleDifferentNote: string | null;
  excludeCertainCharacteristics: TriStateAnswer | null;
  excludeCertainCharacteristicsCriteria: string | null;
  excludeCertainCharacteristicsNote: string | null;
}

export interface GetBeneficiaryIdentification_modelPlan {
  __typename: "ModelPlan";
  id: UUID;
  modelName: string;
  beneficiaries: GetBeneficiaryIdentification_modelPlan_beneficiaries;
}

export interface GetBeneficiaryIdentification {
  modelPlan: GetBeneficiaryIdentification_modelPlan;
}

export interface GetBeneficiaryIdentificationVariables {
  id: UUID;
}
