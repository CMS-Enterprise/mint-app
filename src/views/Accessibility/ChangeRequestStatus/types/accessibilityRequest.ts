/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
