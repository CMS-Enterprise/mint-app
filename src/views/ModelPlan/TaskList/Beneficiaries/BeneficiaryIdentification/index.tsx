import React, { useRef } from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Link, useHistory, useParams } from 'react-router-dom';
import { useMutation, useQuery } from '@apollo/client';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  Fieldset,
  Grid,
  GridContainer,
  IconArrowBack,
  Label,
  Radio
} from '@trussworks/react-uswds';
import { Field, Form, Formik, FormikProps } from 'formik';

import AddNote from 'components/AddNote';
import AskAQuestion from 'components/AskAQuestion';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import Alert from 'components/shared/Alert';
import AutoSave from 'components/shared/AutoSave';
import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
import FieldErrorMsg from 'components/shared/FieldErrorMsg';
import FieldGroup from 'components/shared/FieldGroup';
import MultiSelect from 'components/shared/MultiSelect';
import TextAreaField from 'components/shared/TextAreaField';
import TextField from 'components/shared/TextField';
import GetBeneficiaryIdentification from 'queries/Beneficiaries/getBeneficiaryIndentification';
import {
  GetBeneficiaryIdentification as BeneficiaryIdentificationType,
  GetBeneficiaryIdentification_modelPlan_beneficiaries as BeneficiaryIdentificationFormType
} from 'queries/Beneficiaries/types/GetBeneficiaryIdentification';
import { UpdateModelPlanBeneficiariesVariables } from 'queries/types/UpdateModelPlanBeneficiaries';
import UpdateModelPlanBeneficiaries from 'queries/UpdateModelPlanBeneficiaries';
import { BeneficiariesType, TriStateAnswer } from 'types/graphql-global-types';
import flattenErrors from 'utils/flattenErrors';
import { sortOtherEnum, translateBeneficiariesType } from 'utils/modelPlan';

const BeneficiaryIdentification = () => {
  const { t } = useTranslation('beneficiaries');
  const { t: h } = useTranslation('draftModelPlan');
  const { modelID } = useParams<{ modelID: string }>();

  const formikRef = useRef<FormikProps<BeneficiaryIdentificationFormType>>(
    null
  );
  const history = useHistory();

  const { data } = useQuery<BeneficiaryIdentificationType>(
    GetBeneficiaryIdentification,
    {
      variables: {
        id: modelID
      }
    }
  );

  const {
    id,
    beneficiaries,
    beneficiariesOther,
    beneficiariesNote,
    treatDualElligibleDifferent,
    treatDualElligibleDifferentHow,
    treatDualElligibleDifferentNote,
    excludeCertainCharacteristics,
    excludeCertainCharacteristicsCriteria,
    excludeCertainCharacteristicsNote
  } =
    data?.modelPlan?.beneficiaries || ({} as BeneficiaryIdentificationFormType);

  const modelName = data?.modelPlan?.modelName || '';

  const [update] = useMutation<UpdateModelPlanBeneficiariesVariables>(
    UpdateModelPlanBeneficiaries
  );

  const handleFormSubmit = (
    formikValues: BeneficiaryIdentificationFormType,
    redirect?: 'next' | 'back'
  ) => {
    update({
      variables: {
        id,
        changes: formikValues
      }
    })
      .then(response => {
        if (!response?.errors) {
          if (redirect === 'next') {
            history.push(`/models/${modelID}/task-list/beneficiaries/page-two`);
          } else if (redirect === 'back') {
            history.push(`/models/${modelID}/task-list/`);
          }
        }
      })
      .catch(errors => {
        formikRef?.current?.setErrors(errors);
      });
  };

  const mappedBeneficiariesType = Object.keys(BeneficiariesType)
    .sort(sortOtherEnum)
    .map(key => ({
      value: key,
      label: translateBeneficiariesType(key)
    }));

  const initialValues: BeneficiaryIdentificationFormType = {
    __typename: 'PlanBeneficiaries',
    id: id ?? '',
    beneficiaries: beneficiaries ?? '',
    beneficiariesOther: beneficiariesOther ?? '',
    beneficiariesNote: beneficiariesNote ?? '',
    treatDualElligibleDifferent: treatDualElligibleDifferent ?? null,
    treatDualElligibleDifferentHow: treatDualElligibleDifferentHow ?? '',
    treatDualElligibleDifferentNote: treatDualElligibleDifferentNote ?? '',
    excludeCertainCharacteristics: excludeCertainCharacteristics ?? null,
    excludeCertainCharacteristicsCriteria:
      excludeCertainCharacteristicsCriteria ?? '',
    excludeCertainCharacteristicsNote: excludeCertainCharacteristicsNote ?? ''
  };

  return (
    <>
      <BreadcrumbBar variant="wrap">
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to="/">
            <span>{h('home')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb>
          <BreadcrumbLink asCustom={Link} to={`/models/${modelID}/task-list/`}>
            <span>{h('tasklistBreadcrumb')}</span>
          </BreadcrumbLink>
        </Breadcrumb>
        <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
      </BreadcrumbBar>
      <PageHeading className="margin-top-4 margin-bottom-2">
        {t('heading')}
      </PageHeading>

      <p
        className="margin-top-0 margin-bottom-1 font-body-lg"
        data-testid="model-plan-name"
      >
        <Trans i18nKey="modelPlanTaskList:subheading">
          indexZero {modelName} indexTwo
        </Trans>
      </p>
      <p className="margin-bottom-2 font-body-md line-height-sans-4">
        {h('helpText')}
      </p>

      <AskAQuestion modelID={modelID} />

      <Formik
        initialValues={initialValues}
        onSubmit={values => {
          handleFormSubmit(values, 'next');
        }}
        enableReinitialize
        innerRef={formikRef}
      >
        {(formikProps: FormikProps<BeneficiaryIdentificationFormType>) => {
          const {
            errors,
            handleSubmit,
            setErrors,
            setFieldValue,
            values
          } = formikProps;
          const flatErrors = flattenErrors(errors);
          return (
            <>
              {Object.keys(errors).length > 0 && (
                <ErrorAlert
                  testId="formik-validation-errors"
                  classNames="margin-top-3"
                  heading={h('checkAndFix')}
                >
                  {Object.keys(flatErrors).map(key => {
                    return (
                      <ErrorAlertMessage
                        key={`Error.${key}`}
                        errorKey={key}
                        message={flatErrors[key]}
                      />
                    );
                  })}
                </ErrorAlert>
              )}
              <GridContainer className="padding-left-0 padding-right-0">
                <Grid row gap className="beneficiaries__info">
                  <Grid desktop={{ col: 6 }}>
                    <Form
                      className="margin-top-6"
                      data-testid="beneficiaries-page-one-form"
                      onSubmit={e => {
                        handleSubmit(e);
                      }}
                    >
                      <FieldGroup
                        scrollElement="beneficiaries"
                        error={!!flatErrors.beneficiaries}
                        className="margin-top-4"
                      >
                        <Label htmlFor="beneficiaries-beneficiaries">
                          {t('beneficiaries')}
                        </Label>
                        <FieldErrorMsg>
                          {flatErrors.beneficiaries}
                        </FieldErrorMsg>

                        <Field
                          as={MultiSelect}
                          id="beneficiaries-beneficiaries"
                          name="beneficiaries"
                          options={mappedBeneficiariesType}
                          selectedLabel={t('selectedGroup')}
                          onChange={(value: string[] | []) => {
                            setFieldValue('beneficiaries', value);
                          }}
                          initialValues={initialValues.beneficiaries}
                        />

                        {(values?.beneficiaries || []).includes(
                          'OTHER' as BeneficiariesType
                        ) && (
                          <FieldGroup
                            scrollElement="beneficiariesOther"
                            error={!!flatErrors.beneficiariesOther}
                          >
                            <Label
                              htmlFor="beneficiaries-other"
                              className="text-normal"
                            >
                              {t('beneficiariesOther')}
                            </Label>
                            <FieldErrorMsg>
                              {flatErrors.beneficiariesOther}
                            </FieldErrorMsg>
                            <Field
                              as={TextField}
                              error={flatErrors.beneficiariesOther}
                              id="beneficiaries-other"
                              data-testid="beneficiaries-other"
                              name="beneficiariesOther"
                            />
                          </FieldGroup>
                        )}

                        {(values?.beneficiaries || []).includes(
                          'NA' as BeneficiariesType
                        ) && (
                          <Alert type="info" slim>
                            {t('beneficiariesNA')}
                          </Alert>
                        )}

                        <AddNote
                          id="beneficiaries-note"
                          field="beneficiariesNote"
                        />
                      </FieldGroup>

                      <FieldGroup
                        scrollElement="treatDualElligibleDifferent"
                        error={!!flatErrors.treatDualElligibleDifferent}
                        className="margin-y-4 margin-bottom-8"
                      >
                        <Label htmlFor="beneficiaries-dual-eligibility">
                          {t('dualEligibility')}
                        </Label>
                        <FieldErrorMsg>
                          {flatErrors.treatDualElligibleDifferent}
                        </FieldErrorMsg>
                        <Fieldset>
                          <Field
                            as={Radio}
                            id="beneficiaries-dual-eligibility"
                            name="treatDualElligibleDifferent"
                            label={h('yes')}
                            value="TRUE"
                            checked={
                              values.treatDualElligibleDifferent === 'YES'
                            }
                            onChange={() => {
                              setFieldValue(
                                'treatDualElligibleDifferent',
                                'YES'
                              );
                            }}
                          />

                          {values?.treatDualElligibleDifferent?.includes(
                            'YES' as TriStateAnswer
                          ) && (
                            <FieldGroup
                              className="margin-left-4 margin-y-1"
                              scrollElement="treatDualElligibleDifferentHow"
                              error={
                                !!flatErrors.treatDualElligibleDifferentHow
                              }
                            >
                              <Label
                                htmlFor="beneficiaries-dual-eligibility-how"
                                className="text-normal"
                              >
                                {h('howSo')}
                              </Label>
                              <FieldErrorMsg>
                                {flatErrors.treatDualElligibleDifferentHow}
                              </FieldErrorMsg>
                              <Field
                                as={TextAreaField}
                                className="height-15"
                                error={
                                  flatErrors.treatDualElligibleDifferentHow
                                }
                                id="beneficiaries-dual-eligibility-how"
                                data-testid="beneficiaries-dual-eligibility-how"
                                name="beneficiaries-dual-eligibility-how"
                              />
                            </FieldGroup>
                          )}
                          <Field
                            as={Radio}
                            id="beneficiaries-dual-eligibility-no"
                            name="treatDualElligibleDifferent"
                            label={h('no')}
                            value="FALSE"
                            checked={
                              values.treatDualElligibleDifferent === 'NO'
                            }
                            onChange={() => {
                              setFieldValue(
                                'treatDualElligibleDifferent',
                                'NO'
                              );
                            }}
                          />
                          <Field
                            as={Radio}
                            id="beneficiaries-dual-eligibility-tbd"
                            name="treatDualElligibleDifferent"
                            label={t('beneficiariesOptions.na')}
                            value="TBD"
                            checked={
                              values.treatDualElligibleDifferent === 'TBD'
                            }
                            onChange={() => {
                              setFieldValue(
                                'treatDualElligibleDifferent',
                                'TBD'
                              );
                            }}
                          />
                        </Fieldset>
                        <AddNote
                          id="beneficiaries-dual-eligibility-note"
                          field="treatDualElligibleDifferentNote"
                        />
                      </FieldGroup>

                      <FieldGroup
                        scrollElement="excludeCertainCharacteristics"
                        error={!!flatErrors.excludeCertainCharacteristics}
                        className="margin-y-4 margin-bottom-8"
                      >
                        <Label htmlFor="beneficiaries-exclude">
                          {t('excluded')}
                        </Label>
                        <FieldErrorMsg>
                          {flatErrors.excludeCertainCharacteristics}
                        </FieldErrorMsg>
                        <Fieldset>
                          <Field
                            as={Radio}
                            id="beneficiaries-exclude"
                            name="excludeCertainCharacteristics"
                            label={h('yes')}
                            value="TRUE"
                            checked={
                              values.excludeCertainCharacteristics === 'YES'
                            }
                            onChange={() => {
                              setFieldValue(
                                'excludeCertainCharacteristics',
                                'YES'
                              );
                            }}
                          />

                          {values?.excludeCertainCharacteristics?.includes(
                            'YES' as TriStateAnswer
                          ) && (
                            <FieldGroup
                              className="margin-left-4 margin-y-1"
                              scrollElement="excludeCertainCharacteristicsCriteria"
                              error={
                                !!flatErrors.excludeCertainCharacteristicsCriteria
                              }
                            >
                              <Label
                                htmlFor="beneficiaries-exclude-criteria"
                                className="text-normal"
                              >
                                {t('excludedNestedQuestion')}
                              </Label>
                              <FieldErrorMsg>
                                {
                                  flatErrors.excludeCertainCharacteristicsCriteria
                                }
                              </FieldErrorMsg>
                              <Field
                                as={TextAreaField}
                                className="height-15"
                                error={
                                  flatErrors.excludeCertainCharacteristicsCriteria
                                }
                                id="beneficiaries-exclude-criteria"
                                data-testid="beneficiaries-exclude-criteria"
                                name="beneficiaries-exclude-criteria"
                              />
                            </FieldGroup>
                          )}

                          <Field
                            as={Radio}
                            id="beneficiaries-exclude-no"
                            name="excludeCertainCharacteristics"
                            label={h('no')}
                            value="FALSE"
                            checked={
                              values.excludeCertainCharacteristics === 'NO'
                            }
                            onChange={() => {
                              setFieldValue(
                                'excludeCertainCharacteristics',
                                'NO'
                              );
                            }}
                          />
                          <Field
                            as={Radio}
                            id="beneficiaries-exclude-tbd"
                            name="excludeCertainCharacteristics"
                            label={t('beneficiariesOptions.na')}
                            value="TBD"
                            checked={
                              values.excludeCertainCharacteristics === 'TBD'
                            }
                            onChange={() => {
                              setFieldValue(
                                'excludeCertainCharacteristics',
                                'TBD'
                              );
                            }}
                          />
                        </Fieldset>
                        <AddNote
                          id="beneficiaries-exclude-note"
                          field="excludeCertainCharacteristicsNote"
                        />
                      </FieldGroup>

                      <div className="margin-top-6 margin-bottom-3">
                        <Button type="submit" onClick={() => setErrors({})}>
                          {h('next')}
                        </Button>
                      </div>
                      <Button
                        type="button"
                        className="usa-button usa-button--unstyled"
                        onClick={() => handleFormSubmit(values, 'back')}
                      >
                        <IconArrowBack className="margin-right-1" aria-hidden />
                        {h('saveAndReturn')}
                      </Button>
                    </Form>
                  </Grid>
                </Grid>
              </GridContainer>
              {id && (
                <AutoSave
                  values={values}
                  onSave={() => {
                    handleFormSubmit(formikRef.current!.values);
                  }}
                  debounceDelay={3000}
                />
              )}
            </>
          );
        }}
      </Formik>
      <PageNumber currentPage={1} totalPages={3} className="margin-y-6" />
    </>
  );
};

export default BeneficiaryIdentification;
