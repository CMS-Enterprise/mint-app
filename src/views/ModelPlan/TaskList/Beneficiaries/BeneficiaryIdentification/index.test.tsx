import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';

import GetBeneficiaryIdentification from 'queries/Beneficiaries/getBeneficiaryIndentification';
import { GetBeneficiaryIdentification_modelPlan_beneficiaries as GetBeneficiaryIdentificationType } from 'queries/Beneficiaries/types/GetBeneficiaryIdentification';
import { BeneficiariesType, TriStateAnswer } from 'types/graphql-global-types';

import BeneficiaryIdentification from './index';

const mockData: GetBeneficiaryIdentificationType = {
  __typename: 'PlanBeneficiaries',
  id: '123',
  beneficiaries: ['OTHER' as BeneficiariesType],
  beneficiariesOther: 'other',
  beneficiariesNote: 'note',
  treatDualElligibleDifferent: 'YES' as TriStateAnswer,
  treatDualElligibleDifferentHow: 'This is how',
  treatDualElligibleDifferentNote: 'This is note',
  excludeCertainCharacteristics: 'YES' as TriStateAnswer,
  excludeCertainCharacteristicsCriteria: 'Exclude',
  excludeCertainCharacteristicsNote: 'Note'
};

const beneficiaryMock = [
  {
    request: {
      query: GetBeneficiaryIdentification,
      variables: { id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905' }
    },
    result: {
      data: {
        modelPlan: {
          id: 'ce3405a0-3399-4e3a-88d7-3cfc613d2905',
          modelName: 'My excellent plan that I just initiated',
          opsEvalAndLearning: mockData
        }
      }
    }
  }
];

describe('Model Plan Beneficiaries', () => {
  it('renders without errors', async () => {
    render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/beneficiaries'
        ]}
      >
        <MockedProvider mocks={beneficiaryMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/beneficiaries">
            <BeneficiaryIdentification />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(
        screen.getByTestId('beneficiaries-page-one-form')
      ).toBeInTheDocument();
    });
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[
          '/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/beneficiaries'
        ]}
      >
        <MockedProvider mocks={beneficiaryMock} addTypename={false}>
          <Route path="/models/:modelID/task-list/beneficiaries">
            <BeneficiaryIdentification />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
