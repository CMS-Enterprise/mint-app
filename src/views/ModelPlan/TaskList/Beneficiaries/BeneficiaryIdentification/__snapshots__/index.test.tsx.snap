// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model Plan Beneficiaries matches snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumbs"
    class="usa-breadcrumb usa-breadcrumb--wrap"
  >
    <ol
      class="usa-breadcrumb__list"
    >
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="usa-breadcrumb__list-item"
      >
        <a
          class="usa-breadcrumb__link"
          href="/models/ce3405a0-3399-4e3a-88d7-3cfc613d2905/task-list/"
        >
          <span>
            Model Plan task list
          </span>
        </a>
      </li>
      <li
        aria-current="page"
        class="usa-breadcrumb__list-item usa-current"
      >
        breadcrumb
      </li>
    </ol>
  </nav>
  <h1
    aria-live="polite"
    class="mint-h1 margin-top-4 margin-bottom-2"
    tabindex="-1"
  >
    Beneficiaries
  </h1>
  <p
    class="margin-top-0 margin-bottom-1 font-body-lg"
    data-testid="model-plan-name"
  >
    for 
  </p>
  <p
    class="margin-bottom-2 font-body-md line-height-sans-4"
  >
    If there's a question or field that is not applicable to your model or you don't currently know the answer, you may leave it blank. If you need help, ask a question using the link below.
  </p>
  <div
    class="padding-2 bg-primary-lighter display-flex"
    data-testid="ask-a-question"
  >
    <svg
      class="usa-icon text-primary margin-right-1"
      focusable="false"
      height="1em"
      role="img"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <button
      class="usa-button usa-button--unstyled"
      data-testid="ask-a-question-button"
      type="button"
    >
      Ask a question
    </button>
  </div>
  <div
    class="grid-container padding-left-0 padding-right-0"
    data-testid="gridContainer"
  >
    <div
      class="grid-row grid-gap beneficiaries__info"
      data-testid="grid"
    >
      <div
        class="desktop:grid-col-6"
        data-testid="grid"
      >
        <form
          action="#"
          class="margin-top-6"
          data-testid="beneficiaries-page-one-form"
        >
          <div
            class="usa-form-group margin-top-4"
            data-scroll="beneficiaries"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="beneficiaries-beneficiaries"
            >
              Who are the beneficiaries of this model? Select all that apply.
            </label>
            <div
              class="easi-multiselect maxw-mobile-lg position-relative"
              id="beneficiaries-beneficiaries"
            >
              <div
                class="easi-multiselect__field"
                role="listbox"
              >
                <div
                  class="usa-select maxw-none padding-0"
                >
                  <input
                    class="usa-input padding-1 height-full border-0"
                    placeholder="0 selected"
                    type="search"
                    value=""
                  />
                  <div
                    class="easi-multiselect__controls"
                  >
                    <div
                      class="easi-multiselect__controls-button easi-multiselect__controls-close"
                    >
                      <svg
                        class="usa-icon usa-icon--size-4"
                        focusable="false"
                        height="1em"
                        role="button"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-y-4 margin-bottom-8"
            data-scroll="treatDualElligibleDifferent"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="beneficiaries-dual-eligibility"
            >
              Should dual-eligible beneficiaries be treated differently than non-dually eligibles?
            </label>
            <fieldset
              class="usa-fieldset"
              data-testid="fieldset"
            >
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-dual-eligibility"
                  name="treatDualElligibleDifferent"
                  type="radio"
                  value="TRUE"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-dual-eligibility"
                >
                  Yes
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-dual-eligibility-no"
                  name="treatDualElligibleDifferent"
                  type="radio"
                  value="FALSE"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-dual-eligibility-no"
                >
                  No
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-dual-eligibility-tbd"
                  name="treatDualElligibleDifferent"
                  type="radio"
                  value="TBD"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-dual-eligibility-tbd"
                >
                  Not applicable
                </label>
              </div>
            </fieldset>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="usa-form-group margin-y-4 margin-bottom-8"
            data-scroll="excludeCertainCharacteristics"
          >
            <label
              class="usa-label"
              data-testid="label"
              for="beneficiaries-exclude"
            >
              Should beneficiaries with certain characteristics or enrollments be excluded?
            </label>
            <fieldset
              class="usa-fieldset"
              data-testid="fieldset"
            >
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-exclude"
                  name="excludeCertainCharacteristics"
                  type="radio"
                  value="TRUE"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-exclude"
                >
                  Yes
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-exclude-no"
                  name="excludeCertainCharacteristics"
                  type="radio"
                  value="FALSE"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-exclude-no"
                >
                  No
                </label>
              </div>
              <div
                class="usa-radio"
                data-testid="radio"
              >
                <input
                  class="usa-radio__input"
                  id="beneficiaries-exclude-tbd"
                  name="excludeCertainCharacteristics"
                  type="radio"
                  value="TBD"
                />
                <label
                  class="usa-radio__label"
                  for="beneficiaries-exclude-tbd"
                >
                  Not applicable
                </label>
              </div>
            </fieldset>
            <div
              class="margin-top-4 margin-bottom-8"
            >
              <button
                class="usa-button usa-button usa-button--unstyled"
                data-testid="add-note-toggle"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="usa-icon margin-right-1"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                Add an additional note
              </button>
            </div>
          </div>
          <div
            class="margin-top-6 margin-bottom-3"
          >
            <button
              class="usa-button"
              data-testid="button"
              type="submit"
            >
              Next
            </button>
          </div>
          <button
            class="usa-button usa-button usa-button--unstyled"
            data-testid="button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="usa-icon margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            Save and return to task list
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    class="mint-page-number margin-y-6"
  >
    <span
      class="mint-page-number__page-num"
    >
      Page 1 of 3
    </span>
  </div>
</DocumentFragment>
`;
